cmd/containerExecuteStructureTests.go

```diff
@@ -0,0 +1,98 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+type containerExecuteStructureTestsUtils interface {
+	Stdout(out io.Writer)
+	Stderr(err io.Writer)
+	RunExecutable(e string, p ...string) error
+	Glob(pattern string) (matches []string, err error)
+}
+
+type containerExecuteStructureTestsUtilsBundle struct {
+	*command.Command
+	*piperutils.Files
+}
+
+func newContainerExecuteStructureTestsUtils() containerExecuteStructureTestsUtils {
+	utils := containerExecuteStructureTestsUtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func containerExecuteStructureTests(config containerExecuteStructureTestsOptions, _ *telemetry.CustomData) {
+	utils := newContainerExecuteStructureTestsUtils()
+	err := runContainerExecuteStructureTests(&config, utils)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func findConfigFiles(pattern string, utils containerExecuteStructureTestsUtils) ([]string, error) {
+	files, err := utils.Glob(pattern)
+	if err != nil {
+		return nil, err
+	}
+
+	return files, nil
+}
+
+func runContainerExecuteStructureTests(config *containerExecuteStructureTestsOptions, utils containerExecuteStructureTestsUtils) error {
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	containerStructureTestsExecutable := "container-structure-test"
+	var parameters []string
+	parameters = append(parameters, "test")
+	configFiles, err := findConfigFiles(config.TestConfiguration, utils)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find config files, error: %v", err)
+	}
+	if len(configFiles) == 0 {
+		return errors.New("config files mustn't be missing")
+	}
+	for _, config := range configFiles {
+		parameters = append(parameters, "--config", config)
+	}
+	if config.TestDriver != "" {
+		if config.TestDriver != "docker" && config.TestDriver != "tar" {
+			return fmt.Errorf("test driver %s is incorrect. Possible drivers: docker, tar", config.TestDriver)
+		}
+		parameters = append(parameters, "--driver", config.TestDriver)
+	} else if os.Getenv("ON_K8S") == "true" {
+		parameters = append(parameters, "--driver", "tar")
+	} else {
+		parameters = append(parameters, "--driver", "docker")
+	}
+	if config.PullImage {
+		parameters = append(parameters, "--pull")
+	}
+	if config.TestImage == "" {
+		return errors.New("testImage must not be empty")
+	}
```

Comment:
We can omit this since we'll never reach it when `testImage` is not set due to mandatory flag.